aiogram_dialog/manager/manager_middleware.py

```diff
@@ -13,21 +16,71 @@ def __init__(self, registry: DialogRegistryProto):
         super().__init__()
         self.registry = registry
 
-    async def on_pre_process_message(self, event: Any, data: dict):
+    async def __call__(
+            self,
+            handler: Callable[[Union[Update, DialogUpdateEvent], Dict[str, Any]], Awaitable[Any]],
+            event: Union[Update, DialogUpdateEvent],
+            data: Dict[str, Any],
+    ) -> Any:
+
+        if isinstance(event, DialogUpdateEvent):
+            content = event
+            data.update(
+                {'bot': event.bot,
+                 'event_from_user': event.from_user,
+                 'event_chat': event.chat,
+                 }
+            )
+        else:
+            content = detect_content_type(event)
+
         data[MANAGER_KEY] = ManagerImpl(
-            event=event,
+            event=content,
             registry=self.registry,
             data=data,
         )
 
-    on_pre_process_callback_query = on_pre_process_message
-    on_pre_process_aiogd_update = on_pre_process_message
-    on_pre_process_my_chat_member = on_pre_process_message
+        try:
+            return await handler(event, data)
+        finally:
+            manager: DialogManager = data.pop(MANAGER_KEY)
+            await manager.close_manager()
 
-    async def on_post_process_message(self, _, result, data: dict):
-        manager: DialogManager = data.pop("dialog_manager")
-        await manager.close_manager()
 
-    on_post_process_callback_query = on_post_process_message
-    on_post_process_aiogd_update = on_post_process_message
-    on_post_process_my_chat_member = on_post_process_message
+def detect_content_type(update):
```

Comment:
0_o

can we simplify it?

Reply:
I was thinking about add to Update property that return event,
or iterate fields that isinstanse(field,  TelegramObject) and get first not None field